dll/win32/kernel32_vista/DllMain.c

```diff
@@ -1,6 +1,9 @@
 
 #include "k32_vista.h"
 
+char __roscompat_descriptor__;
+char* p__roscompat_descriptor__ = &__roscompat_descriptor__;
```

Comment:
To ensure it is included in the final binary I assume?
For msvc we could simply do `#pragma comment(linker, "/include:__roscompat_descriptor__")`
For gcc, we can either include a function with the attribute `__attribute__((constructor))` that references this table,
or move the stubs to a library, and add the flag `--whole-archive` to that library

Reply:
Wouldn't `__attribute__((used))` be sufficient when compiling with GCC, or does that not work as expected for Windows?

Reply:
It's actually not needed.